tests/test/functions.rs

```diff
@@ -61,6 +62,9 @@ fn function_implement_fn_traits() {
             struct Ty { }
 
             trait Clone { }
+
+            impl Clone for Ty { }
+
             opaque type MyOpaque: Clone = Ty;
```

Comment:
The last two tests in this set are supposed to flounder, but instead return "no more solutions". Maybe the floundering behavior has purposefully changed since this test was written?

Reply:
I think something like the following should be added here https://github.com/rust-lang/chalk/blob/0b391f8eed657ac23abad15e5d6ae9a5c7852f11/chalk-solve/src/clauses.rs#L423-L426
```rust
if trait_datum.is_non_enumerable_trait() && /* trait does not appear in opaque type bound */ {
    return Err(Floundered);
}
```

Reply:
New output looks correct here